geonode/qgis_server/views.py

```diff
@@ -380,172 +380,53 @@ def geotiff(request, layername, access_token=None):
-    for param, value in params.iteritems():
-        url += param + '=' + value + '&'
-
-    logger.debug(url)
-
-    result = urllib2.urlopen(url)
-    result_list = result.readlines()
-    return HttpResponse(
-        ''.join(result_list).replace('\n', ''), content_type='text/xml')
-
-
-def wfs_describe_feature_type(params):
-    logger.debug('WFS Describe Feature Type')
-    qgis_server = QGIS_SERVER_CONFIG['qgis_server_url']
-
-    layer = Layer.objects.get(typename=params.pop('TYPENAME'))
-    params['LAYERS'] = layer.name
-    try:
-        qgis_layer = QGISServerLayer.objects.get(layer=layer)
-    except ObjectDoesNotExist:
-        msg = 'No QGIS Server Layer for existing layer %s' % layer.name
-        logger.debug(msg)
-        raise Http404(msg)
-
-    basename, _ = os.path.splitext(qgis_layer.base_layer_path)
-
-    params['map'] = basename + '.qgs'
-
-    url = qgis_server + '?'
-    for param, value in params.iteritems():
-        url += param + '=' + value + '&'
-
-    logger.debug(url)
-
-    result = urllib2.urlopen(url)
-    result_list = result.readlines()
-    return HttpResponse(
-        ''.join(result_list).replace('\n', ''), content_type='text/xml')
-
+def qgis_server_request(request):
+    """View to proxy OGC request to QGIS Server."""
+    # Make a copy of the query string with capital letters for the key.
+    query = request.GET or request.POST
+    params = {
+        param.upper(): value for param, value in query.iteritems()}
 
-def wms_get_feature_info(params):
-    logger.debug('WMS GetFeatureInfo')
+    # 900913 is deprecated
+    if params.get('SRS') == 'EPSG:900913':
+        params['SRS'] = 'EPSG:3857'
+    if params.get('CRS') == 'EPSG:900913':
+        params['CRS'] = 'EPSG:3857'
 
-    qgis_server = QGIS_SERVER_CONFIG['qgis_server_url']
+    # As we have one QGIS project per layer, we don't support GetCapabilities
+    # for now without any layer.
+    if params.get('REQUEST') == 'GetCapabilities':
+        if not params.get('LAYERS') or params.get('TYPENAME'):
+            # Todo, we need to fix this. Or having another LAYER parameter.
+            return HttpResponse('GetCapabilities is not supported yet.')
 
-    layer = Layer.objects.get(typename=params.pop('LAYERS'))
-    params['LAYERS'] = layer.name
-    params['QUERY_LAYERS'] = layer.name
-    logger.debug(params['QUERY_LAYERS'])
-    try:
-        qgis_layer = QGISServerLayer.objects.get(layer=layer)
-    except ObjectDoesNotExist:
-        msg = 'No QGIS Server Layer for existing layer %s' % layer.name
-        logger.debug(msg)
-        raise Http404(msg)
+    # We have some shortcuts here.
+    if params.get('SERVICE') == 'WMS':
+        if params.get('REQUEST') == 'GetLegendGraphic':
+            if not params.get('LAYERS'):
+                raise Http404('LAYERS is not found for a GetLegendGraphic')
+            layer = get_object_or_404(Layer, typename=params.get('LAYER'))
+            return legend(request, layername=layer.name)
```

Comment:
@lucernae As we removing the legend, how can I get the legend path?
If not easy, I can forward the request to QGIS Server.